treeherder/webapp/api/performance_data.py

```diff
@@ -2,16 +2,82 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, you can obtain one at http://mozilla.org/MPL/2.0/.
 
+import json
+import time
+
 from django.core.cache import cache
 from rest_framework import viewsets
 from rest_framework.response import Response
 from rest_framework.decorators import list_route
 from rest_framework_extensions.etag.decorators import etag
 
-
+from treeherder.model import models
 from treeherder.model.derived.jobs import JobsModel
 from treeherder.webapp.api.utils import (with_jobs)
+from treeherder.webapp.api import serializers as th_serializers
+
+
+class PerformanceSignatureViewSet(viewsets.ViewSet):
+
+    def list(self, request, project):
+
+        repository = models.Repository.objects.get(name=project)
+
+        # TODO: support other parameters like interval, etc
+
+        signature_data = models.PerformanceDatum.objects.filter(
+            repository=repository).select_related(
+                'signature', 'signature__option_collection',
+                'signature__platform').values_list(
+                    'signature__uuid',
+                    'signature__option_collection__option_collection_hash',
+                    'signature__platform__platform', 'signature__suite',
+                    'signature__test', 'signature__extra_properties').distinct()
+
+        ret = {}
+        for (uuid, option_collection_hash, platform, suite, test,
+             extra_properties) in signature_data:
+            ret[uuid] = {
+                'option_collection_hash': option_collection_hash,
+                'machine_platform': platform,
+                'suite': suite,
+                'test': test,
+            }
+            ret[uuid].update(json.loads(extra_properties))
+
+        return Response(ret)
+
+class PerformanceDatumViewSet(viewsets.ViewSet):
+
+    def list(self, request, project):
+        repository = models.Repository.objects.get(name=project)
+
+        try:
+            signatures = request.QUERY_PARAMS.getlist("signatures")
+        except Exception:
+            return Response("incorrect parameters", 400)
 
+        # TODO: support other parameters like interval, etc
+
+        # FIXME: could we make this more efficient by using a serializer?
+        ret = {}
+        for signature in signatures:
+            signature_obj = models.PerformanceSignature.objects.get(uuid=signature)
+            datums = models.PerformanceDatum.objects.filter(repository=repository,
```

Comment:
This will generate 2 queries per signature, which is probably too much.
You can obtain more or less the same with
```python
datum_list = models.PerformanceDatum.objects.filter(repository=repository, signature__uuid__in=signatures).order_by('signature__uuid', 'push_timestamp')
```